Index: run-tests.php
===================================================================
RCS file: /repository/php-src/run-tests.php,v
retrieving revision 1.223
diff -u -r1.223 run-tests.php
--- run-tests.php	18 Jul 2005 00:19:28 -0000	1.223
+++ run-tests.php	25 Jul 2005 12:28:31 -0000
@@ -926,6 +926,12 @@
 		$query_string = '';
 	}
 
+	if (!empty($section_text['ENV'])) {
+		foreach (explode("\n", $section_text['ENV']) as $env) {
+			($env = trim($env)) and putenv($env);
+		}
+	}
+
 	putenv("REDIRECT_STATUS=1");
 	putenv("QUERY_STRING=$query_string");
 	putenv("PATH_TRANSLATED=$tmp_file");
@@ -951,7 +957,7 @@
 		putenv("CONTENT_TYPE=");
 		putenv("CONTENT_LENGTH=");
 
-		$cmd = "$php$pass_options$ini_settings -f \"$tmp_file\" $args 2>&1";
+		$cmd = "$php$pass_options$ini_settings < \"$tmp_file\" $args 2>&1";
 	}
 
 	if ($DETAILED) echo "
@@ -968,6 +974,13 @@
 //	$out = `$cmd`;
 	$out = system_with_timeout($cmd);
 
+	if (!empty($section_text['ENV'])) {
+		foreach (explode("\n", $section_text['ENV']) as $env) {
+			$env = explode('=', $env);
+			($env = trim($env)) and putenv($env[0] .'=');
+		}
+	}
+
 	@unlink($tmp_post);
 
 	// Does the output match what is expected?
